File: _assets/scripts/normalizers/biomes_normalize_old.lua
return function (data)
    local data_attr = python.as_attrgetter(data)
    if data_attr.__contains__("minecraft:biome") then
        return
    end
    local biome_identifier = nil
    for index, key in python.enumerate(data) do
        biome_identifier = key
        break
    end
    assert(biome_identifier ~= nil, "dictionary is empty; cannot find biome identifier")
    local output = python.builtins.dict()
    output["format_version"] = data[biome_identifier]["format_version"]
    output["minecraft:biome"] = python.builtins.dict()
    output["minecraft:biome"]["components"] = data[biome_identifier]
    output["minecraft:biome"]["description"] = python.builtins.dict()
    output["minecraft:biome"]["description"]["identifier"] = biome_identifier
    python.as_attrgetter(output["minecraft:biome"]["components"]).__delitem__("format_version")
    return output
end

File: _assets/scripts/normalizers/biomes_normalize_old.py
from typing import Any, TypedDict

__all__ = ["biomes_normalize_old"]

input_biome_typed_dict = TypedDict("input_biome_typed_dict", {
    "format_version": str,
})

input_typed_dict = dict[str,input_biome_typed_dict]

output_description_typed_dict = TypedDict("output_description_typed_dict", {
    "identifier": str,
})

output_biome_typed_dict = TypedDict("output_biome_typed_dict", {
    "components": Any,
    "description": output_description_typed_dict,
})

output_typed_dict = TypedDict("output_typed_dict", {
    "format_version": str,
    "minecraft:biome": output_biome_typed_dict,
})

def biomes_normalize_old(data:input_typed_dict|output_typed_dict) -> output_typed_dict|None:
    if "minecraft:biome" in data:
        return None
    biome_identifier = list(data.keys())[0]
    output:output_typed_dict = {
        "format_version": data[biome_identifier]["format_version"],
        "minecraft:biome": {
            "components": data[biome_identifier],
            "description": {
                "identifier": biome_identifier
            }
        }
    }
    del output["minecraft:biome"]["components"]["format_version"]
    return output
